src/core/messageRouter.ts

```diff
@@ -0,0 +1,27 @@
+import TelegramMessage from '../classes/TelegramMessage';
+import UseCase from '../interfaces/useCase';
+import TelegramMessageType from '../classes/TelegramMessageType';
+
+export default class MessageRouter {
+  private useCases: UseCase[] = [];
+
+  registerUseCase(useCase: UseCase): void {
+    this.useCases.push(useCase);
+  }
+
```

Comment:
Jetzt hast du ja doch wieder Kommunikation in beide Richtungen ? Wenn sich die use Cases anmelden müssen?

Reply:
Die Use Cases selbst melden sich nicht an und wissen nichts von der ganzen Message Handling-Infrastruktur, sondern hier wird nur die Use Case-Instanz übergeben.
Siehe: https://github.com/JaydenUlysses/Olaf-the-coolest-of-Bots/blob/0902973a350f9486d2a174ed5a8baa69bb8dbef5/src/core/olaf.ts#L36